OpenRA.Mods.Common/AI/HackyAI.cs

```diff
@@ -535,6 +545,23 @@ void ITick.Tick(Actor self)
 
 			foreach (var b in builders)
 				b.Tick();
+
+			Sync.CheckSyncUnchanged(World, () =>
+			{
+				foreach (var t in tickModules)
+					if (t.IsTraitEnabled())
+						t.BotTick(this);
+			});
+
+			// Make sure we tick after all of the bot modules so that we don't introduce an additional tick delay
+			World.AddFrameEndTask(IssueOrders);
```

Comment:
We tick the modules ourselves just above, so the FrameEndTask is now redundant &rarr; the body of `IssueOrders` can be moved directly here.